protoc_plugin/lib/grpc_generator.dart

```diff
@@ -224,15 +234,15 @@ class _GrpcMethod {
     out.addBlock(
         '$_clientReturnType $_dartName($_argumentType request, {${GrpcServiceGenerator._callOptions} options}) {',
         '}', () {
-      final requestStream =
-          _clientStreaming ? 'request' : '$_stream.fromIterable([request])';
+      final createCall = _clientStreaming || _serverStreaming
+          ? '\$createStreamingCall'
```

Comment:
You can use `r'...'` then you don't need to escape `$`

Reply:
Made explicit codegen for each case (unary/streaming), as it was quite unreadable.